app/src/main/java/com/keylesspalace/tusky/StatusListActivity.kt

```diff
@@ -67,6 +79,64 @@ class StatusListActivity : BottomSheetActivity(), HasAndroidInjector {
         }
     }
 
+    override fun onCreateOptionsMenu(menu: Menu): Boolean {
+        val tag = hashtag
+        if (kind == Kind.TAG && tag != null) {
+            lifecycleScope.launch {
+                try {
+                    val tagEntity = mastodonApi.tag(tag).await()
+                    menuInflater.inflate(R.menu.view_hashtag_toolbar, menu)
+                    followTagItem = menu.findItem(R.id.action_follow_hashtag)
+                    unfollowTagItem = menu.findItem(R.id.action_unfollow_hashtag)
+                    followTagItem?.isVisible = tagEntity.following == false
+                    unfollowTagItem?.isVisible = tagEntity.following == true
+                    followTagItem?.setOnMenuItemClickListener { followTag() }
+                    unfollowTagItem?.setOnMenuItemClickListener { unfollowTag() }
+                } catch (exception: Exception) {
+                    Log.w("Failed to query tag #$tag", exception)
+                }
+            }
+        }
+
+        return super.onCreateOptionsMenu(menu)
+    }
+
+    private fun followTag(): Boolean {
+        val tag = hashtag
+        if (tag != null) {
+            val context = this
+            lifecycleScope.launch {
+                try {
+                    mastodonApi.followTag(tag).await()
+                    followTagItem?.isVisible = false
+                    unfollowTagItem?.isVisible = true
+                } catch (_: Exception) {
+                    Toast.makeText(context, "Error following #$tag", Toast.LENGTH_SHORT).show()
```

Comment:
the error message should be translateable, please additionally log the exception (same for unfollwing ofc)